cloudquery/sdk/plugin/plugin.py

```diff
@@ -29,10 +29,40 @@ class SyncOptions:
     backend_options: BackendOptions = None
 
 
+@dataclass
+class BuildTarget:
+    os: str = None
+    arch: str = None
+
+
+@dataclass
+class Dockerfile:
+    path: str = None
+    build_targets: List[BuildTarget] = None
+
+
+@dataclass
+class Options:
+    dockerfiles: List[Dockerfile] = None
+    team_name: str = None
+    plugin_kind: str = None
+
+
 class Plugin:
-    def __init__(self, name: str, version: str) -> None:
+    def __init__(self, name: str, version: str, opts: Options = None) -> None:
         self._name = name
         self._version = version
+        self._opts = Options() if opts is None else opts
+        if self._opts.dockerfiles is None:
+            self._opts.dockerfiles = [
+                Dockerfile("Dockerfile", [
```

Comment:
I suggest to keep the defaults as what we have now (only `linux/amd64` and `linux/arm64`, see https://github.com/cloudquery/cloudquery/blob/3e2b5a4db5157dc58ed3507c4b44f95c7fde5be4/.github/workflows/release_plugin.yml#L93

Other platforms might be tricker to build